packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1758,9 +1758,24 @@ mixin WidgetInspectorService {
   }
 
   List<Object> _getChildrenSummaryTree(String? diagnosticsNodeId, String groupName) {
-    final DiagnosticsNode? node = toObject(diagnosticsNodeId) as DiagnosticsNode?;
+    final Object? theObject = toObject(diagnosticsNodeId);
     final InspectorSerializationDelegate delegate = InspectorSerializationDelegate(groupName: groupName, summaryTree: true, service: this);
-    return _nodesToJson(node == null ? const <DiagnosticsNode>[] : _getChildrenFiltered(node, delegate), delegate, parent: node);
+
+    // TODO(polina-c): remove this, when DevTools stops sending DiagnosticsNode.
```

Comment:
nit: file a bug on DevTools for this and mention it here to make it easier to track.

Reply:
This is tracked as subtask of https://github.com/flutter/devtools/issues/3951. added to comment.